src/pam/securemem.rs

```diff
@@ -1,19 +1,30 @@
 //! Routines for "secure" memory operations; i.e. data that we need to send to Linux-PAM and don't
 //! want any copies to leak (that we would then need to zeroize).
-use std::slice;
+use std::{
+    alloc::{self, Layout},
+    mem,
+    ptr::NonNull,
+    slice,
+};
 
-pub struct PamBuffer(*mut u8);
+const SIZE: usize = super::sys::PAM_MAX_RESP_SIZE as usize;
+const ALIGN: usize = mem::align_of::<u8>();
 
-impl PamBuffer {
-    const SIZE: usize = super::sys::PAM_MAX_RESP_SIZE as usize;
+pub struct PamBuffer(NonNull<[u8; SIZE]>);
+
+fn layout() -> Layout {
```

Comment:
I initially tried to put this into a `const` to have the `from_size_align` checks run at compile time but `{Result,Option}::unwrap` is not a const function so that didn't work